src/lib/server/bitcoin-nodeless.ts

```diff
@@ -32,27 +32,30 @@ export function isZPubValid(zpub: string): boolean {
 }
 
 export async function generateDerivationIndex(): Promise<number> {
-	const res = await collections.runtimeConfig.findOneAndUpdate(
-		{ _id: 'bitcoinNodeless' },
-		{
-			$inc: { 'data.derivationIndex': 1 as never },
-			$set: { updatedAt: new Date() }
-		},
-		{ returnDocument: 'before' }
-	);
-
-	if (!res.value) {
-		throw new Error('Failed to increment derivation index');
+	let index = runtimeConfig.bitcoinNodeless.derivationIndex;
+	
+	for (let attempts = 0; attempts < 10; attempts++) {
+		const address = bip84Address(runtimeConfig.bitcoinNodeless.publicKey, index);
+		const isUsed = await isAddressUsed(address).catch(() => false);
```

Comment:
This catch silences a potentially unrelated issue (e.g. API rate-limit); we should at least log the failure.